modules/sand/src/main/ts/ephox/sand/api/SandHTMLElement.ts

```diff
@@ -16,7 +18,9 @@ const isPrototypeOf = (x: any): x is HTMLElement => {
   // undefined scope later triggers using the global window.
   const scope: Window | undefined = Resolve.resolve('ownerDocument.defaultView', x);
 
-  return sandHTMLElement(scope).prototype.isPrototypeOf(x);
+  // TINY-7374: We can't rely on looking at the owner window HTMLElement as the element may have
+  // been constructed in a different window and then appended to the current window document.
+  return Type.isObject(x) && (sandHTMLElement(scope).prototype.isPrototypeOf(x) || /^\[object HTML\w*Element\]$/.test(getPrototypeOf(x).toString()));
```

Comment:
Instead of doing the main test, and then falling back to the regex, why not only do the regex?

```suggestion
  return Type.isObject(x) && /^\[object HTML\w*Element\]$/.test(getPrototypeOf(x).toString());
```

Reply:
The other case is a guaranteed version so it 100% should not be removed. This also keeps it consistent with the pattern used in `Type`.

Reply:
I should add this also gives us some protection in case something isn't caught by the prototype name comparison.